lib/query-emitter.js

```diff
@@ -41,23 +43,30 @@ QueryEmitter.prototype._open = function() {
   this._defaultCallback = function(err) {
     if (err) emitter.onError(err);
   };
-  emitter.stream.on('data', function(data) {
-    if (data.error) {
-      return emitter.onError(data.error);
-    }
-    emitter._update(data);
-  });
-  emitter.stream.on('end', function() {
-    emitter.destroy();
+
+  emitter.streams.forEach(function(stream) {
+    stream.on('data', function(data) {
+      if (data.error) {
+        return emitter.onError(data.error);
+      }
+      emitter._update(data);
+    });
+    stream.on('end', function() {
+      emitter.destroy();
+    });
   });
+
   // Make sure we start polling if pollInterval is being used
   this._flushPoll();
 };
 
 QueryEmitter.prototype.destroy = function() {
   clearTimeout(this._pollDebounceId);
   clearTimeout(this._pollIntervalId);
-  this.stream.destroy();
+
+  this.streams.forEach(function(stream) {
+    stream.destroy();
+  });
```

Comment:
(Alec) This may cause recursive calls to the same emitter's destroy, where this triggers another stream's destroy, which causes this emitter's stream be destroyed again.

He suggest doing something like
```js
var stream;
while (stream = streams.pop()) {
  stream.destroy();
}
```